python/cuml/fil/fil.pyx

```diff
@@ -0,0 +1,232 @@
+        treelite_params.output_class = output_class
+        treelite_params.threshold = self.threshold
+        treelite_params.algo = self.algo
+        cdef forest_t* forest_pointer =\
+            <forest_t*><size_t> model.handle.value
+        cdef cumlHandle* handle_ =\
+            <cumlHandle*><size_t>self.handle.getHandle()
+        cdef uintptr_t model_ptr = model.handle.value
+        self.forest_data = from_treelite(handle_[0],
+                                         forest_pointer,
+                                         <ModelHandle> model_ptr,
+                                         &treelite_params)
+        return self
+
+    def __cdel__(self):
+        cdef cumlHandle* handle_ =\
+            <cumlHandle*><size_t>self.handle.getHandle()
+        free(handle_[0],
+             self.forest_data)
+        return self
+
+
+class FIL(Base):
+    """
+    Parameters
+    ----------
+    model : the model information in the treelite format
+       loaded from a saved model using the treelite API
+       https://treelite.readthedocs.io/en/latest/treelite-api.html
+    output_class: boolean
+       True or False
+    algo : 0 = NAIVE, 1 = TREE_REORG, 2 = BATCH_TREE_REORG
+    threshold : threshold is used to for classification
+       if output == OUTPUT_CLASS, else it is ignored
+    """
+    def __init__(self,
+                 algo=0,
+                 threshold=0.0,
+                 handle=None):
+        super(FIL, self).__init__(handle)
+        self.algo_type = algo
+        self.threshold = threshold
+        self._impl = FIL_impl(algo,
+                              threshold,
+                              self.handle)
+
+    def predict(self, X, preds=None):
+        """
+        Predicts the labels for X on the model that was loaded
+        using from_treelite function.
+        Parameters
+        ----------
+        X : array-like (device or host) shape = (n_samples, n_features)
+           Dense matrix (floats or doubles) of shape (n_samples, n_features).
+           Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+           ndarray, cuda array interface compliant array like CuPy
+        preds: NumPy or cuDF dataframe
+           Dense vector (int) of shape (n_samples, 1)
+        Returns
+        ----------
+        labels predicted by the model
+        """
+        return self._impl.predict(X, preds)
+
+    def from_treelite(self, model, output_class):
+        """
+        Creates a FIL model using the treelite model
+        passed to the function.
+        Parameters
+        ----------
+        model : the model information in the treelite format
+           loaded from a saved model using the treelite API
+           https://treelite.readthedocs.io/en/latest/treelite-api.html
+        output_class: boolean
+           True or False
+        """
+        return self._impl.from_treelite(model, output_class)
+
+
+def from_treelite_direct(model, algo=0, output_class=True,
```

Comment:
`from_treelite()` is a better name. As two `from_treelite()` functions are defined in different scopes, there is no name conflict.

Reply:
maybe `fil_from_treelite`? or `load_from_treelite`?

Reply:
Is it possible to move `cdef` definitions to a different Python file/module? In this way, we could expose more concise names to the user, e.g. `from_treelite`.

Reply:
I thought this would be nice too, so I gave it a shot in a prototype. But I found it created a fair amount of code duplication and more confusing layout, plus I ran into a number of annoying compilation issues. I'd love to revisit post-0.9 as the complexity grows, but I think it would be hard to do nicely by 0.9. Also, we've made the names shorter in the new rev ;)